Gauge enhancements #749

Kinlaw opened the pull request:
Adds the new boolean property fullCircle to gauge which when set to true, will allow a user to create a full-circle gauge. The default value is false.

Also adds the new property startingAngle to gauge which allows the user to select the starting angle in radians. The default is set to -Pi/2.

aendrew commented:
@Kinlaw This looks terrific. Could you please do the following?

1. Squash all your commits please.
2. Remove c3.js from your commits because it gets built automatically during releases (and including it reduces the likelihood your code will merge without issue).
3. Most importantly, please add a unit test for *each piece of new functionality* — this helps us ensure that the PR works, and that future changes don't break it. Thank you!

Kinlaw commented:
Thanks for reviewing Andrew!

1) I squashed the commits.
2) Removed the changes to c3.js, c3.min.js, c3.css.
3) The c3.css changes were a result of the grunt build so I removed those changes.
4) I added unit tests

Please let me know if there's anything else that you need from me!

aendrew commented:
Disregard the failing check above, it's due to some weirdness in how something's being linted.

I've merged this into `dev`; thanks for the PR @Kinlaw! :D :tada: :shipit:

Kinlaw commented:
Sweet!

aendrew commented:
@Kinlaw I'm noticing an issue with your test, mind taking a look? https://travis-ci.org/masayuki0812/c3/builds/110638853#L541

masayuki0812 commented:
This has been merged into master as 0.4.11. Thanks!
